scanner/providers/implementations/themoviedatabase.py

```diff
@@ -138,26 +140,88 @@ def to_studio(self, company: dict[str, Any]) -> Studio:
 			},
 		)
 
+	async def get_best_image(self, item, lng, key):
+		"""
+		Retrieves the best available images for a item based on localization.
+
+		Args:
+			item (dict): A dictionary containing item information, including images and language details.
+			lng (Language): The preferred language for the images.
+			key (str): The key to access the images in the item dictionary. (e.g. "posters", "backdrops", "logos")
+		Returns:
+			list: A list of images, prioritized by localization, original language, and any available image.
+		"""
+		# Order images by size and vote average
+		item["images"][key] = sorted(
+			item["images"][key],
+			key=lambda x: (x.get("vote_average", 0), x.get("width", 0)),
+			reverse=True,
+		)
+
+		# Step 1: Try to get localized images
+		localized_images = [
+			image
+			for image in item["images"][key]
+			if image.get("iso_639_1") == lng.language
+		]
+
+		# Step 2: If no localized images, try images in the original language
+		if not localized_images:
+			localized_images = [
+				image
+				for image in item["images"][key]
+				if image.get("iso_639_1") == item["original_language"]
+			]
+
+		# Step 3: If still no images, use any available images
+		if not localized_images:
+			localized_images = item["images"][key]
+
+		# Corner case: If there are no images at all, call TMDB images API.
+		# Although doing another API call is not ideal, this would only be called very rarely.
+		if not localized_images:
+			logger.debug(
+				"[Fallback] No images found for %s %s. Calling TMDB images API.",
+				item["media_type"],
+				item["id"],
+			)
+			images = await self.get(
+				f"{item['media_type']}/{item['id']}/images",
+			)
+			localized_images = sorted(
+				images[key],
+				key=lambda x: x.get("vote_average", 0),
+				reverse=True,
+			)
```

Comment:
I was able to get 100% of my library items with posters by adding this corner case (of 400, 1 item was a corner case).

I know doing a extra request is bad, and this is also increases the code complexity of this method because it needs to be `async`. If this is too much, I can remove.

Reply:
Yes!

Reply:
For those cases, is the `poster_path` of the base item empty or does it contain a vaild image?

Reply:
Using `poster_path` seems to be enough, I`ll remove the extra call